src/vs/platform/terminal/common/xterm/shellIntegrationAddon.ts

```diff
@@ -309,6 +309,7 @@ export class ShellIntegrationAddon extends Disposable implements IShellIntegrati
 	constructor(
 		private _nonce: string,
 		private readonly _disableTelemetry: boolean | undefined,
+		private _initialCwd: string | undefined,
```

Comment:
This moves some reconnection logic where it should not be imo. Why do we need to do this here but not for commands that get restored? This is after reloading the window, see commands are all intact:

![image](https://github.com/user-attachments/assets/7412e09d-10e6-4b8e-bb22-c78999320d4c)

The cwd sequence(s) should be in the serialized content that gets "replayed", just like the commands? This is the code that generates the replay:

https://github.com/microsoft/vscode/blob/7a3d738bbb52a0222cbd97277b07d93113b79139/src/vs/platform/terminal/node/ptyService.ts#L876-L889